andela/ah-rambo-backend
# #167573212 Implement Login Feature

#### What does this PR do?
Implement login route

#### Description of Task proposed in this pull request?
- add login route
- test login route
- create user model, migration and seeds file
- add a `findUser` helper function
- add a password helper function
- fix houndCI issue

#### How should this be manually tested (Quality Assurance)?
- check into the branch by running `git checkout ft-login-167573212`
- create a `.env` file, copy the contents of the `.env.sample` file into it and update the fields and also your database details.
- run `npm db:rollback` to migrate and seed into the database.
- run `npm run start:dev` to start the server
- open [postman](https://www.getpostman.com/downloads/) and use seeded to the database to test e.g 
`POST http://localhost:9000/api/v1/sessions/create`
`{`
                `userLogin: 'JhayX'`
                `password: 'incorrect'`
`}`

#### What are the relevant pivotal tracker stories?
- Story ID:  [#167573212](https://www.pivotaltracker.com/story/show/167573212)

#### Any background context you want to add (Operations Impact)?
- run `npm install` to install all dependencies
- update the `.env` file with your database connection string
- run `npm run db:rollback` to create tables in the database and seed 

#### What I have learned working on this feature:
- I learnt how to mock server error test with sinon

#### Screenshots: 
Successful Login:
<img width="1434" alt="Screenshot 2019-07-30 at 6 56 22 PM" src="https://user-images.githubusercontent.com/47576037/62153214-fb0d5080-b2fb-11e9-8769-5bcec0e11d80.png">

Unsuccessful login:
<img width="1440" alt="Screenshot 2019-07-30 at 6 59 33 PM" src="https://user-images.githubusercontent.com/47576037/62154103-f9448c80-b2fd-11e9-83a1-dadf136a6dd7.png">
